src/crawlee/storages/_dataset.py

```diff
@@ -107,13 +107,17 @@ async def open(
         configuration = service_locator.get_configuration() if configuration is None else configuration
         storage_client = service_locator.get_storage_client() if storage_client is None else storage_client
 
+        client_opener = storage_client.create_dataset_client(id=id, name=name, alias=alias, configuration=configuration)
+        additional_cache_key = storage_client.get_additional_cache_key(configuration=configuration)
+
         return await service_locator.storage_instance_manager.open_storage_instance(
             cls,
             id=id,
             name=name,
             alias=alias,
-            configuration=configuration,
-            client_opener=storage_client.create_dataset_client,
+            client_opener=client_opener,
+            storage_client_type=storage_client.__class__,
```

Comment:
Maybe just a string literal would also be good, definitely not a request, probably subjective